### PR TITLE
[BugFix] Parametric collectors

### PR DESCRIPTION
**Pull request by matteobettini**

### CODE REVIEW

**Review by vmoens — COMMENTED** [review]
Could you add a test (make sure it's a short one, ie just one worker for the mp collectors)

File: test/test_env.py [hunk 1]
```diff
@@ -1393,14 +1393,14 @@ def test_nested_reward(self, envclass):
...
-        if envclass == "NestedRewardEnv":
+        if envclass == "NestedCountingEnv":
             assert ("next", "data", "done") in next_state.keys(True)
             assert ("next", "data", "states") in next_state.keys(True)
             assert ("next", "data", "reward") in next_state.keys(True)
         assert ("next", *env.done_key) in next_state.keys(True)
         assert ("next", *env.reward_key) in next_state.keys(True)
 
-        check_env_specs(env)
+        # check_env_specs(env)
```
Comment (matteobettini) [inline test/test_env.py:1403]: this needs #1268 fixed to work

**Comment by matteobettini** [comment]
I have added the test (if it s too long feel free to cut some parts of it).
It uses an inproved version of NestedCountinEnv which will help us test nested envs with an added dimensionality for nested keys (just like MARL)

File: test/test_collector.py [hunk 2]
```diff
@@ -1346,6 +1355,99 @@ def test_reset_heterogeneous_envs():
...
+            frames_per_batch=frames_per_batch,
+            total_frames=100,
+            device="cpu",
+        )
+
+        for _td in ccollector:
+            break
+        ccollector.shutdown()
+
+        # assert ("data","reward") not in td.keys(True) # this can be activates once step_mdp is fixed for nested keys
```
Comment (matteobettini) [inline test/test_collector.py:1437]: this needs https://github.com/pytorch/rl/issues/1268 fixed to work

**Comment by matteobettini** [comment]
I have added more general tests for nested envs and specific collector tests

File: test/test_env.py [hunk 3]
```diff
@@ -1420,7 +1453,7 @@ def test_nested_reward(self, envclass):
         MockBatchedLockedEnv,
         MockBatchedUnLockedEnv,
         MockSerialEnv,
-        NestedRewardEnv,
+        # NestedCountingEnv,
```
Comment (matteobettini) [inline test/test_env.py:1456]: this needs https://github.com/pytorch/rl/issues/1268 fixed to be uncommented

File: test/test_collector.py (hunk 2 above)
Comment (vmoens) [reply test/test_collector.py:1437]: Got it, afaic we can merge it and uncomment later

File: test/test_env.py (hunk 1 above)
Comment (vmoens) [reply test/test_env.py:1403]: ditto

File: test/test_env.py (hunk 3 above)
Comment (vmoens) [reply test/test_env.py:1456]: ditto

**Review by vmoens — APPROVED** [review]
LGTM thanks!

**Comment by matteobettini** [comment]
Ok mergable for me

**Comment by vmoens** [comment]
Let's reduce the test time and we're good to go!

**Comment by matteobettini** [comment]
Is it still too high after the commit named slimmer test?

**Pull request merged**